typing/typecore.ml

```diff
@@ -4406,13 +4397,9 @@ and type_expect_
       [Nolabel, sbody]) ->
       if txt = "extension.once" && not (Language_extension.is_enabled Unique) then
         raise (Typetexp.Error (loc, Env.empty, Unsupported_extension Unique));
-      let mode = if mode_cross env ty_expected then mode_max
-      else begin
-        let mode = mode_once expected_mode in
-        submode ~loc ~env ~reason:Other mode.mode expected_mode;
-        mode
-      end
-      in
+      let expected_mode = expect_mode_cross env ty_expected expected_mode in
+      let mode = mode_once expected_mode in
+      submode ~loc ~env ~reason:Other mode.mode expected_mode;
```

Comment:
This line is equivalent to:
```ocaml
submode ~loc ~env ~reason:Other Value.once expected_mode;
```
but much less clear. And if you reorder it with the line above then you can do:
```ocaml
let expected_mode = expect_mode_cross env ty_expected expected_mode in
submode ~loc ~env ~reason:Other Value.once expected_mode;
let expected_mode = mode_once expected_mode in
```
and avoid the need to have variables called both `mode` and `expected_mode`.